components/ai_chat/browser/ai_chat_api.h

```diff
@@ -32,6 +32,10 @@ class AIChatAPI {
                        data_completed_callback,
                    api_request_helper::APIRequestHelper::DataReceivedCallback
                        data_received_callback = base::NullCallback());
+  // Clears all in-progress requests
+  // TODO(nullhook): Keep track of in-progress requests and cancel them
+  // individually. This would be useful to keep some in-progress requests alive.
```

Comment:
Perhaps this TODO should be in the .cc. Also when would it be good to keep in-progress requests alive? If they were generated before the user clicks "clear", I would think we should abort them?

Reply:
Moved to .cc

Keeping in-progress requests alive would be useful if a user can send multiple messages and can individually stop generating one. This was earlier versions of AIChat I believe and perhaps doesn't apply now. However, a scenario I thought about was for quick navigation: A user navigates back and forth quickly and we might need to cache in-progress requests within a certain threshold to keep the requests alive?